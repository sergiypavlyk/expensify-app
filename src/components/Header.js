import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" className={({ isActive }) => isActive ? 'is-active' : ''} exact="true">Dashboard</NavLink>
    <NavLink to="/create" className={({ isActive }) => isActive ? 'is-active' : ''}>Create Expanse</NavLink>
    <NavLink to="/edit" className={({ isActive }) => isActive ? 'is-active' : ''}>Edit Expanse</NavLink>
    <NavLink to="/help" className={({ isActive }) => isActive ? 'is-active' : ''}>Help</NavLink>
  </header>
)

export default Header;
