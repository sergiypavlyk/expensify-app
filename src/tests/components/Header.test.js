import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  )
  expect(screen.getByRole('heading').textContent).toBe('Expensify')
})

