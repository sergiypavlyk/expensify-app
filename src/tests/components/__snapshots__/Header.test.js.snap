// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    className={[Function]}
    exact="true"
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    className={[Function]}
    to="/create"
  >
    Create Expanse
  </NavLink>
  <NavLink
    className={[Function]}
    to="/edit"
  >
    Edit Expanse
  </NavLink>
  <NavLink
    className={[Function]}
    to="/help"
  >
    Help
  </NavLink>
</header>
`;
